services/pull/patch.go

```diff
@@ -60,7 +60,7 @@ var patchErrorSuffices = []string{
 
 // TestPatch will test whether a simple patch will apply
 func TestPatch(pr *issues_model.PullRequest) error {
-	ctx, _, finished := process.GetManager().AddContext(graceful.GetManager().HammerContext(), fmt.Sprintf("TestPatch: Repo[%d]#%d", pr.BaseRepoID, pr.Index))
+	ctx, _, finished := process.GetManager().AddContext(graceful.GetManager().HammerContext(), fmt.Sprintf("TestPatch: %s", pr))
```

Comment:
normally an object (pr) can not be converted into string(%s) ...

Reply:
https://github.com/go-gitea/gitea/blob/d11768b5c8b0dbb0919765a8b0f85e150f3dcb8f/models/issues/pull.go#L258-L261

https://github.com/go-gitea/gitea/blob/d11768b5c8b0dbb0919765a8b0f85e150f3dcb8f/modules/log/colors.go#L297-L301

https://github.com/go-gitea/gitea/blob/d11768b5c8b0dbb0919765a8b0f85e150f3dcb8f/models/issues/pull.go#L227-L239

Reply:
Actually, in my mind, I think it's better to drop the color support to keep code simple.

Because these logs are mainly for help & support purpose. Users just collect the logs and then post them to the support channel, then the colors are lost in the end.

(Just my trivial opinion, since this is being discussed)

Reply:
It's still going to require a lot of complicated code to account for the possibility of whether the repo is loaded or not.

The color doesn't add that much more complication. 

If we make the move to slog/structured logging we can begin the move to simplify things but we'd still need to do similar things there too.

Reply:
About "whether the repo is loaded or not.", another trivial question: does it need to do `log.Error` before every `return err` or `return false`? Some returns have the log.Error now, while some others still don't.

Reply:
The question is always whether such logging would ever be or is helpful. 

I've tried to add logging where I think it would be helpful. If there is a line where you think I've missed something please comment on it. 

I've added a defer log.Trace to testPR which probably accounts for a lot of things.

We will have to go through our logging at some point and consider when things are best to log. There are a few things that we probably need to do in general, in that our errors need more context and probably need equivalents of structured logging. (This could include stacktraces as we've discussed before.)